xonsh/foreign_shells.py

```diff
@@ -580,11 +580,13 @@ def load_foreign_aliases(shells=None, config=None, issue_warning=True):
     for shell in shells:
         shell = ensure_shell(shell)
         _, shaliases = foreign_shell_data(**shell)
+        shaliases = {} if shaliases is None else shaliases
         for alias in set(shaliases) & set(xonsh_aliases):
-            del shaliases[alias]
-            print('aliases: alias {!r} of shell {!r} '
-                  'tries to override xonsh alias, '
-                  'xonsh wins!'.format(alias, shell['shell']),
-                  file=sys.stderr)
+            if alias in shaliases:
```

Comment:
You don't have to ask. You are right the test is superfluous. Did you just manually integrate it into your PR?